render/gles2/color.c

```diff
@@ -0,0 +1,241 @@
+// OpenGL Color conversion
+
+#define _XOPEN_SOURCE 500 // for strdup
+
+#include <GLES3/gl32.h>
+
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <time.h>
+
+#include <wlr/types/wlr_list.h>
+#include <wlr/render/color.h>
+
+#include <lcms2.h>
+
+#include "render/gles2.h"
+
+// GL3+ or GL_EXT_texture_norm16 on ES 3.2
+#define GL_RGBA16 0x805B
+
+struct lut {
+	GLuint tex_id;
+	struct wlr_color_config *input, *output;
+	cmsHTRANSFORM transform;
+};
+
+#define LUTS_MAX 100
+
+static struct wlr_list luts;
+
+static int color_compare(void *av, void *bv) {
+	struct wlr_color_config *a = av, *b = bv;
```

Comment:
You're comparing a struct containing a pointer to another struct containing a pointer. This won't ever work.
You should compare either the ICC profile bytes, the ICC profile parameters (better) or the colorspace (if provided).

Reply:
The strcmp is there - I'm not quite sure what you're saying.

Reply:
Didn't see it.